Automattic/wp-calypso
# Marketplace: clean up and setup paid plugins purchase flow

#### Changes proposed in this Pull Request

* Totally removed `productGroupSlug` in the scope of marketplace as its not used
* Removed all static product references as we are using the API for getting marketplace products https://github.com/Automattic/wp-calypso/blob/ee72bb988e93f14c7e49c59e6d441e60d3a864b2/client/data/marketplace/use-wpcom-plugins-query.ts#L33
* Removes code and files not used anymore. We kept this code until we decided the best path forward. 
* Renames `marketplace-plugin-upload-status` to `marketplace-plugin-install` since it handles all plugin installation flows.
* Uses a simple redirect to checkout flow for buying a subscription which in turn activates the product in question in the backend  using `Store_Product_Marketplace_Plugins->activate`.
* **Disallows removing the business plan when buying a paid plugin on a lower tier**
* **Adds paid plugin flow to `/marketplace/test/` page, by redirecting the user to the checkout with plan (if needed) and subscription**

#### Testing instructions



<img width="1318" alt="SS 2021-12-10 at 19 27 07" src="https://user-images.githubusercontent.com/12430020/145616080-d098d5be-2353-42c6-a67b-3ebbf139c9db.png">


**Scenario 1: wporg plugin installation regressions**
- Install any plugin in your site
- After landing in the Marketplace thank you page confirm that the plugin is installed
- Check for regressions

**Scenario 2: wpcom plugin installation**
- Go to `/marketplace/test/[domain]`
- Click `Pay & Install Woocommerce Subscription`
- After completing the checkout with the monthly subscription you should be redirected to the Marketplace thank you page
- Confirm that Woocommerce Subscriptions is installed / activated in your site

Repeat the above for a lower tier plan (eg `free`), `business` plan should be added in the checkout without being able to remove it as long as the plugin subscription is not removed.

**Scenario 3: checkout thank you page regressions**
- Upgrade your plan
- Check for any regressions in thank you page



Fixes #55575
